src/controllers/Flashcards.js

```diff
@@ -0,0 +1,99 @@
+const Flashcard = require('../models/Flashcard');
+const { StatusCodes } = require('http-status-codes');
+
+const getAllFlashcards = async (req, res) => {
+    const flashcards = await Flashcard.find({}).sort('createdAt');
+    res.status(StatusCodes.OK).json({ flashcards, count: flashcards.length });
+};
+
+const getUserFlashcards = async (req, res) => {
+    const flashcards = await Flashcard.find({ createdBy: req.user.userId }).sort('createdAt');
+    res.status(StatusCodes.OK).json({ flashcards, count: flashcards.length });
+};
+
+const getFlashcard = async (req, res) => {
+    const { 
+        user: { userId }, 
+        params: { id: flashcardId } 
+    } = req;
+
+    console.log("User ID:", userId);
+    console.log("Flashcard ID:", flashcardId);
+
+    try {
+        const flashcard = await Flashcard.findOne({
+            _id: flashcardId, createdBy: userId
+        });
+    
+        if (!flashcard) {
+            console.log(`No flashcard with id ${flashcardId}`);
+            return res.status(StatusCodes.NOT_FOUND).json({ msg: "Flashcard not found" });
+        }
+    
+        console.log('Flashcard 11111', flashcard);
+        res.status(StatusCodes.OK).json({ flashcard });
+    } catch (error) {
+        console.error(error);
+        res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({ msg: "Internal server error" });
+    }
+    
+};
+
+const createFlashcard = async (req, res) => {
+    console.log('Request body:', req.body);
+    console.log('User Object:', req.user); // !!! undefined
+
+    req.body.createdBy = req.user.userId;
+    const flashcard = await Flashcard.create(req.body);
+    res.status(StatusCodes.CREATED).json({ flashcard });
+};
+
+const updateFlashcard = async (req, res) => {
+    const {
+        body: { topic, author, question, answer, createdBy },
+        user: { userId },
+        params: { id: flashcardId }
+    } = req;
+
+    if (topic === '' || author === '' || question === '' || answer === '' || createdBy === '') {
+        console.log('Fields cannot be empty!');
```

Comment:
[Debug Code & Incorrect Error Handling]
This is debug code, but also you should've returned a 400 (Bad Request) error here.
Theres also a question of if you want to handle it this way with every field required?
Generally, the practice is to require the complete document on POST, but not PATCH. On PATCH they can give only part of the document and you'll only change those parts.

I think its fine to require it all here, but we should ensure they get a 400 with a reasonable message here if they don't give you all the fields.

Reply:
Replaced debug code with 400 error here. I could also remove the if statement from here, if you think it's not reasonable.